File: backend/services/core/src/main/java/app/mnema/core/user/config/UserClientConfig.java
package app.mnema.core.user.config;

import org.springframework.boot.context.properties.EnableConfigurationProperties;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.web.client.RestClient;

@Configuration
@EnableConfigurationProperties(UserClientProps.class)
public class UserClientConfig {

    @Bean
    public RestClient userRestClient(UserClientProps props) {
        return RestClient.builder()
                .baseUrl(props.baseUrl())
                .build();
    }
}

File: backend/services/auth/src/main/kotlin/app/mnema/auth/config/UserClientConfig.kt
package app.mnema.auth.config

import org.springframework.boot.context.properties.EnableConfigurationProperties
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.web.client.RestClient

@Configuration
@EnableConfigurationProperties(UserClientProps::class)
class UserClientConfig {
    @Bean
    fun userRestClient(props: UserClientProps): RestClient =
        RestClient.builder()
            .baseUrl(props.baseUrl)
            .build()
}
